### PR TITLE
Automating CEPH-83574724 clone subvolume with full data

### PR DESCRIPTION
**Pull request by AmarnatReddy**
Automating CEPH-83574724 clone subvolume with full data
Polarion : https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574724

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-I48GHV/
IBM-C : https://159.23.92.24/job/rhceph-test-execution-manual/44/console
Signed-off-by: Amarnath K <amk@amk.remote.csb>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/snapshot_clone/clone_subvolume_full_vol.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            [client1],
+            fuse_mounting_dir_2,
+            extra_params=f" -r {clonevol_path.strip()}",
+        )
+        client1.exec_command(
+            sudo=True, cmd=f"diff -qr {kernel_mounting_dir_1} {fuse_mounting_dir_2}"
+        )
+        return 0
+    except Exception as e:
+        log.info(e)
```
Comment (julpark-rh) [inline tests/cephfs/snapshot_clone/clone_subvolume_full_vol.py:127]: @AmarnatReddy did not we decide to use `log.error(e)` for Exceptions?

Comment (julpark-rh) [reply tests/cephfs/snapshot_clone/clone_subvolume_full_vol.py]: LGTM

**Comment by yogesh-mane** [comment]
@AmarnatReddy , In logs I see , smallfiles 'total data =     4.258 GiB' . Could we increase it to more than 5.3 Gib?

**Comment by AmarnatReddy** [comment]
> @AmarnatReddy , In logs I see , smallfiles 'total data = 4.258 GiB' in redhat cloud & 'total data = 3.302 GiB' in ibm cloud . Could we increase it to more than 5.3 Gib?

Hi Yogesh,
I have increased the treads from 3 to 5 now the volume is getting filled completely.
we are seeing dis quota exceeded for the volume also
https://159.23.92.24/job/rhceph-test-execution-manual/49

**Comment by yogesh-mane** [comment]
@AmarnatReddy , could we have check of how much data filled in subvolume?

**Comment by AmarnatReddy** [comment]
> @AmarnatReddy , could we have check of how much data filled in subvolume?

Hi Yogesh,
I have updated script to validate the bytes used.
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-R7WF4H/

**Review by yogesh-mane — DISMISSED** [review]
LGTM

**Review by yogesh-mane — APPROVED** [review]

**Pull request merged**